react/display-name false positive when using React.memo

`eslint-plugin-react@7.12.1`

We have a functional component that is exported with `React.memo`
```jsx
import React from 'react'
import { string } from 'prop-types'

function Component({ world }) {
  return <div>Hello {world}</div>
}

Component.propTypes = {
  world: string,
}

export default React.memo(Component)
```

This triggers the `react/display-name` error after the update from `7.11.1`
```
[...]/Component.jsx
  12:16  error  Component definition is missing display name  react/display-name
```
The exported components `displayName` is `Memo(Component)`.

Without `React.memo` it recognizes the `displayName`.
```jsx
export default Component
```

--- a/lib/util/Components.js
+++ b/lib/util/Components.js
@@ -479,7 +479,7 @@ function componentRule(rule, context) {
         const isArgument = node.parent && node.parent.type === 'CallExpression'; // Arguments (callback, etc.)
         // Attribute Expressions inside JSX Elements (<button onClick={() => props.handleClick()}></button>)
         const isJSXExpressionContainer = node.parent && node.parent.type === 'JSXExpressionContainer';
-        if (node.parent && this.isPragmaComponentWrapper(node.parent)) {
+        if (isFunction && node.parent && this.isPragmaComponentWrapper(node.parent)) {
           return node.parent;
         }
         // Stop moving up if we reach a class or an argument (like a callback)
@@ -620,7 +620,9 @@ function componentRule(rule, context) {
       if (!utils.isPragmaComponentWrapper(node)) {
         return;
       }
-      components.add(node, 2);
+      if (node.arguments.length > 0 && astUtil.isFunctionLikeExpression(node.arguments[0])) {
+        components.add(node, 2);
+      }
     },
 
     ClassExpression: function(node) {

--- a/tests/lib/rules/display-name.js
+++ b/tests/lib/rules/display-name.js
@@ -437,6 +437,21 @@ ruleTester.run('display-name', rule, {
       createElement("a");
     `,
     parser: 'babel-eslint'
+  }, {
+    code: `
+      import React from 'react'
+      import { string } from 'prop-types'
+
+      function Component({ world }) {
+        return <div>Hello {world}</div>
+      }
+
+      Component.propTypes = {
+        world: string,
+      }
+
+      export default React.memo(Component)
+    `
   }],
 
   invalid: [{

